common/src/main/java/com/wynntils/handlers/inventory/InventoryHandler.java

```diff
@@ -0,0 +1,449 @@
+                }
+            }
+        }
+
+        class ThrowFromHeld extends CaptureHeld {
+            private final AbstractContainerMenu menu;
+            private final ItemStack heldStack;
+
+            public ThrowFromHeld(AbstractContainerMenu menu, ItemStack heldStack) {
+                this.menu = menu;
+                this.heldStack = heldStack;
+            }
+
+            @Override
+            protected void tryDispatch(ItemStack newHeldStack) {
+                if (!(newHeldStack.isEmpty() || areItemsSimilar(heldStack, newHeldStack))) return;
+                int amount = heldStack.getCount() - newHeldStack.getCount();
+                if (amount <= 0) return;
+                WynntilsMod.postEvent(new InventoryInteractionEvent(
+                        menu, new InventoryInteraction.ThrowFromHeld(newHeldStack.copyWithCount(amount))));
+            }
+        }
+
+        class ThrowFromSlot extends CaptureSlot {
+            private final ItemStack slotStack;
+
+            public ThrowFromSlot(AbstractContainerMenu menu, int slotNum, ItemStack slotStack) {
+                super(menu, slotNum);
+                this.slotStack = slotStack;
+            }
+
+            @Override
+            protected void tryDispatch(ItemStack newSlotStack) {
+                if (!(newSlotStack.isEmpty() || areItemsSimilar(slotStack, newSlotStack))) return;
+                int amount = slotStack.getCount() - newSlotStack.getCount();
+                if (amount <= 0) return;
+                WynntilsMod.postEvent(new InventoryInteractionEvent(
+                        menu, new InventoryInteraction.ThrowFromSlot(slotNum, newSlotStack.copyWithCount(amount))));
+            }
+        }
+
+        class Transfer extends CaptureSlot {
+            private final ItemStack slotStack;
+
+            public Transfer(AbstractContainerMenu menu, int slotNum, ItemStack slotStack) {
+                super(menu, slotNum);
+                this.slotStack = slotStack;
+            }
+
+            @Override
+            protected void tryDispatch(ItemStack newSlotStack) {
+                if (!(newSlotStack.isEmpty() || areItemsSimilar(slotStack, newSlotStack))) return;
+                int amount = slotStack.getCount() - newSlotStack.getCount();
+                if (amount <= 0) return;
+                WynntilsMod.postEvent(new InventoryInteractionEvent(
+                        menu, new InventoryInteraction.Transfer(slotNum, newSlotStack.copyWithCount(amount))));
+            }
+        }
+    }
+
+    private static final class PendingInteractionEntry {
+        private final PendingInteraction interaction;
+        private final long expiryTime;
+
+        private PendingInteractionEntry(PendingInteraction interaction) {
+            this.interaction = interaction;
+            this.expiryTime = System.currentTimeMillis() + 500L;
```

Comment:
Please move this "magic" numbers to a constants i the top-level handler class instead.